port/network_mbedtls_wrapper.c

```
@@ -170,6 +189,21 @@ IoT_Error_t iot_tls_connect(Network *pNetwork, TLSConnectParams *params) {
     }
 
     /* Parse client private key... */
+#ifdef CONFIG_AWS_IOT_USE_HARDWARE_SECURE_ELEMENT
+    if (pNetwork->tlsConnectParams.pDevicePrivateKeyLocation[0] == '#') {
+        int8_t slot_id = pNetwork->tlsConnectParams.pDevicePrivateKeyLocation[1] - '0';
+        if (slot_id < 0 || slot_id > 9) {
+            ESP_LOGE(TAG, "Invalid ATECC608A slot ID.");
+            ret = NETWORK_PK_PRIVATE_KEY_PARSE_ERROR;
+        } else {
+            ESP_LOGD(TAG, "Using ATECC608A private key from slot %d", slot_id);
+            ret = atca_mbedtls_pk_init(&(tlsDataParams->pkey), slot_id);
+            if (ret != 0) {
+                ESP_LOGE(TAG, " failed !  atca_mbedtls_pk_init returned %02x", ret);
```

This should return an appropriate error code